test/extensions/filters/http/decompressor/decompressor_filter_test.cc

```diff
@@ -299,6 +300,88 @@ TEST_P(DecompressorFilterTest, ExplicitlyEnableAdvertiseAcceptEncoding) {
                       "br,mock" /* expected_accept_encoding */);
 }
 
+TEST_P(DecompressorFilterTest, ExplicitlyEnableAdvertiseAcceptEncodingOnlyOnce) {
+  TestScopedRuntime scoped_runtime;
```

Comment:
these 4 new tests look like they are almost identical and it's work for the reader to see how they differ.

Can you factor that out with a parameterized helper method in DecompressorFilterTest to make it easier to see what is varying across tests?

Reply:
Fixed.